src/Microsoft.DotNet.Core.Build.Tasks/LockFileCache.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using NuGet.Common;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.Tasks
+{
+    internal class LockFileCache
```

Comment:
So I guess are tasks executed in the same process? What if they were in separate assembly load contexts? 

These questions are for curiosity only.